hack/jenkins/test-history/gen_html.py

```diff
@@ -143,36 +145,67 @@ def gen_html(data, prefix, exact_match=False):
     html.append('</html>')
     return '\n'.join(html), (okay, unstable, failed, skipped)
 
-def gen_suite_overview(suites):
-    html = html_header()
-    for suite, (okay, unstable, failed, skipped) in sorted(suites.iteritems()):
-        html.append('<a class="suite-link" href="suite-{}.html">'.format(suite))
+def gen_metadata_links(suites):
+    html = []
+    for (name, target), (okay, unstable, failed, skipped) in sorted(suites.iteritems()):
+        html.append('<a class="suite-link" href="{}">'.format(target))
         html.append('<span class="total okay">{}</span>'.format(okay))
         html.append('<span class="total unstable">{}</span>'.format(unstable))
         html.append('<span class="total failed">{}</span>'.format(failed))
-        html.append(suite)
+        html.append(name)
         html.append('</a>')
-    html.append('</body>')
-    html.append('</html>')
-    return '\n'.join(html)
+    return html
 
-if __name__ == '__main__':
-    prefix = ''
-    if len(sys.argv) == 2:
-        prefix = sys.argv[1]
-    with open('tests.json') as f:
+def main(args):
+    parser = argparse.ArgumentParser()
+    parser.add_argument('--suites', action='store_true',
+                        help='output test results for each suite')
+    parser.add_argument('--prefixes',
+                        help='comma-separated list of suite prefixes to create pages for')
+    parser.add_argument('--output-dir', required=True,
+                        help='where to write output pages')
+    parser.add_argument('--input', required=True,
+                        help='JSON test data to read for input')
+    options=parser.parse_args(args)
+
+    with open(options.input) as f:
         data = json.load(f)
-    if len(sys.argv) == 3:
-        assert sys.argv[1] == 'gensuites'
+
+    if options.prefixes:
+        # the empty prefix means "all tests"
+        options.prefixes = [''] + options.prefixes.split(',')
```

Comment:
Instead of forcing this every time I believe it will work to change the flag to `e2e,,soak`